pkgs/development/compilers/graalvm/community-edition/default.nix

```diff
@@ -15,8 +15,8 @@ let
   */
   graalvm11-ce-release-version = "22.0.0.2";
   graalvm17-ce-release-version = "22.0.0.2";
-  graalvm11-ce-dev-version = "22.2.0-dev-20220401_1942";
-  graalvm17-ce-dev-version = "22.2.0-dev-20220401_1942";
+  graalvm11-ce-dev-version = "22.2.0-dev-20220415_1945";
+  graalvm17-ce-dev-version = "22.2.0-dev-20220415_1945";
```

Comment:
This field seems to not being updated automatically, so I had to set a new version by hand.

CC @tricktron.

Reply:
BTW, the `graalvm17-ce-dev-version` was updated in PR #168816 however the value here wasn't updated because, as I said, it seems the update script didn't update it automatically.

This is why I changed it here but the `graalvm17-ce-sources.json` has no changes.

Reply:
Hm, strange. How did you run the update script? The current logic is the following:

0. currentVersion = graalvmXX-ce-version depending on platform. That means on `aarch64-darwin` it is always `graalvmXX-ce-dev-version`, while on linux it is always `graalvmXX-ce-version`.
1. newVersion = Get the latest version from https://api.github.com/repos/graalvm/graalvm-ce-builds/releases/latest or newest dev version from https://api.github.com/repos/graalvm/graalvm-ce-dev-builds/releases/latest
2. Compare `currentVersion` with `newVersion`.
3.  If the `newVersion` is newer, then update all the sources AND update the `graalvmXX-version` string in `default.nix`.

In summary, to update to a new version you just need to run `nix-shell maintainers/scripts/update.nix --argstr package graalvmXX-ce` and it will find the new version and update the sources.json. If you however, already manually updated the `graalvmXX-version` in default.nix it won't find the new version and therefore won't update the sources.json file. Now for this special case, or the case where you always want to update the version, I really like the `forceUpdate` flag.

However, I have not yet figured out how to correctly pass the forceUpdate flag:
`nix-shell maintainers/scripts/update.nix --argstr package graalvm17-ce --argstr forceUpdate true` did NOT work.

Reply:
> Hm, strange. How did you run the update script? The current logic is the following:

`nix-shell maintainers/scripts/update.nix --argstr package graalvm17-ce` after changing the value of the `forceUpdate` flag in the code to `true`.

> However, I have not yet figured out how to correctly pass the forceUpdate flag:
nix-shell maintainers/scripts/update.nix --argstr package graalvm17-ce --argstr forceUpdate true did NOT work.

Yeah, this doesn't work sadly. It doesn't seem that `maintainers/scripts/update.nix` has any way to pass a custom flag to it (I think it is doable to do it though, but it will need some changes on it I think).

Changing the value manually is ugly, but works fine for now.

Reply:
@trickton Do you want to investigate the issue in this PR or do you think it is fine if we merge this one as-is and fix it in another PR?

Reply:
@thiagokokada Let's merge and fix it in another pr. I currently can't reproduce this, so I'd be glad if you open a new issue and tell me the exact steps to get the above-mentioned behaviour. Then I can have look at it in more detail.

Reply:
Ok.

I think we can test this better once a new version of GraalVM is released.